src/System.IO.Pipelines.Networking.Tls/ApplicationProtocols.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.IO.Pipelines.Networking.Tls.Internal.Sspi;
+
+namespace System.IO.Pipelines.Networking.Tls
+{
+    public static class ApplicationProtocols
+    {
+        private static readonly byte[] _http1 = {0x68, 0x74, 0x74, 0x70, 0x2f, 0x31, 0x2e, 0x31}; //("http/1.1")
+        private static readonly byte[] _spdy1 = {0x73, 0x70, 0x64, 0x79, 0x2f, 0x31}; //("spdy/1")
+        private static readonly byte[] _spdy2 = {0x73, 0x70, 0x64, 0x79, 0x2f, 0x32}; //("spdy/2")
+        private static readonly byte[] _spdy3 = {0x73, 0x70, 0x64, 0x79, 0x2f, 0x33}; //("spdy/3")
+
+        private static readonly byte[] _traversalUsingRelaysAroundNAT =
+        {
+            0x73, 0x74, 0x75, 0x6E, 0x2E, 0x74, 0x75, 0x72,
+            0x6E
+        }; //("stun.turn")
+
+        private static readonly byte[] _natDiscoveryUsingSessionTraversalUtilitiesforNAT =
+        {
+            0x73, 0x74, 0x75, 0x6E, 0x2E,
+            0x6e, 0x61, 0x74, 0x2d, 0x64, 0x69, 0x73, 0x63, 0x6f, 0x76, 0x65, 0x72, 0x79
+        }; //("stun.nat-discovery")
+
+        private static readonly byte[] _http2overTLS = {0x68, 0x32}; //("h2")
+        private static readonly byte[] _http2overTCP = {0x68, 0x32, 0x63}; //("h2c")
+        private static readonly byte[] _webRTCMediaAndData = {0x77, 0x65, 0x62, 0x72, 0x74, 0x63}; //("webrtc")
+
+        private static readonly byte[] _confidentialWebRTCMediaAndData =
+        {
+            0x63, 0x2d, 0x77, 0x65, 0x62, 0x72, 0x74, 0x63
+        }; //("c-webrtc")
+
+        private static readonly byte[] _Ftp = {0x66, 0x74, 0x70}; //("ftp")
+
+        private static readonly byte[][] _allProtocols =
```

Comment:
What determined the ordering of this list?  If it's guidance from an RFC, attribute it.  If it's conjecture, provide insight as to how you conjectured it.

To help keep diffs clean under reordering, leave the members one item per line.  And please end multiline comma separated structures with a separator.

```C#
// [an explanation of this ordering]
private static readonly byte[][] s_allProtocols =
{
    s_http1,
    s_spdy1,
    ...
    s_confidentialWebRtcMediaAndData,
    s_ftp,
};
```